commands_test.go

```diff
@@ -2251,6 +2251,75 @@ var _ = Describe("Commands", func() {
 			Expect(v).To(Equal("c"))
 		})
 
+		It("should LCS", func() {
+			err := client.MSet(ctx, "key1", "ohmytext", "key2", "mynewtext").Err()
+			Expect(err).NotTo(HaveOccurred())
+
+			lcs, err := client.LCS(ctx, &redis.LCSQuery{
+				Key1: "key1",
+				Key2: "key2",
+			}).Result()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(lcs.MatchString).To(Equal("mytext"))
+
+			lcs, err = client.LCS(ctx, &redis.LCSQuery{
+				Key1: "nonexistent_key1",
+				Key2: "key2",
+			}).Result()
+
+			Expect(err).NotTo(HaveOccurred())
+			Expect(lcs.MatchString).To(Equal(""))
+
+			lcs, err = client.LCS(ctx, &redis.LCSQuery{
+				Key1: "key1",
+				Key2: "key2",
+				Len:  true,
+			}).Result()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(lcs.MatchString).To(Equal(""))
+			Expect(lcs.Len).To(Equal(int64(6)))
+
+			lcs, err = client.LCS(ctx, &redis.LCSQuery{
+				Key1: "key1",
+				Key2: "key2",
+				Idx:  true,
+			}).Result()
+			Expect(err).NotTo(HaveOccurred())
+			Expect(lcs.MatchString).To(Equal(""))
+			Expect(lcs.Len).To(Equal(int64(6)))
+			Expect(lcs.Matches).To(Equal([]redis.LCSMatchedPosition{
+				{
+					Key1:     redis.LCSPosition{Start: 4, End: 7},
+					Key2:     redis.LCSPosition{Start: 5, End: 8},
+					MatchLen: 0,
+				},
+				{
+					Key1:     redis.LCSPosition{Start: 2, End: 3},
+					Key2:     redis.LCSPosition{Start: 0, End: 1},
+					MatchLen: 0,
+				},
+			}))
+
+			lcs, err = client.LCS(ctx, &redis.LCSQuery{
+				Key1:         "key1",
+				Key2:         "key2",
+				Idx:          true,
```

Comment:
Yes, it is possible to respond with errors. The lcs command has multiple situations where errors can occur: 
1. key is not a string type
2. command parameter error
3. data is too large
4. insufficient memory
...
We just need to verify that the LCS API can handle errors normally without worrying about the specific error types. 
Are you interested? You can try to add some tests.

Reply:
@monkey92t As the struct itself will validate the key type and command params so I guess that case is redundant as we're not writing validators in general

How do you suggest we test the data being too large?
Something like the following?
```
largeString := strings.Repeat("x", 11000000)
err := client.MSet(ctx, "key1", "oh"+largeString, "key2", "my"+largeString).Err()
Expect(err).To(HaveOccurred())
lcs, err := client.LCS(ctx, &redis.LCSQuery{
    Key1: "key1",
    Key2: "key2",
}).Result()
Expect(err).To(HaveOccurred())

```

Reply:
We don't need to test for memory overflow, and such tests are very unstable because we can't predict the size of the memory. We only need to test for incorrect data types to verify if the API handles errors correctly.

exp:
```shell
> set key1 "mytest"
> lpush key2 one
> lcs key1 key2
```

Reply:
`key2` is not of string type, and the lcs command cannot process it, so it will return an error.

Reply:
Understood, will add these tests

Reply:
@monkey92t Skipping the following error as it will be avoided by using readType in args
![image](https://user-images.githubusercontent.com/70265851/227446911-e20924e8-c431-4def-8e31-9c5aed2ffe0d.png)

Reply:
Yes, we just need to verify if go-redis can handle errors without focusing on specific error types. Redis-server may increase or decrease its error responses, and we are not writing unit tests for redis-server.